units/generate_tasks.go

```diff
@@ -170,6 +202,26 @@ func (j *generateTasksJob) generate(ctx context.Context, t *task.Task) error {
 
 // handleError return mongo.ErrNoDocuments if another job has raced, the passed in error otherwise.
 func (j *generateTasksJob) handleError(pp *model.ParserProject, v *model.Version, handledError error) error {
+	// Get task again, to exit nil if another generator finished, which caused us to error.
+	// Checking this again here makes it very unlikely that there is a race, because both
+	// `t.GeneratedTasks` checks must have been in between the racing generator's call to
+	// save the config and set the task's boolean.
+	t, err := task.FindOneId(j.TaskID)
+	if err != nil {
+		return errors.Wrapf(err, "error finding task %s", j.TaskID)
+	}
+	if t == nil {
+		return errors.Errorf("unable to find task %s", j.TaskID)
+	}
+	if t.GeneratedTasks {
+		grip.Debug(message.Fields{
+			"message": "handledError encountered task that already generating, nooping",
+			"task":    t.Id,
+			"version": t.Version,
+		})
+		return mongo.ErrNoDocuments
```

Comment:
we should just return nil here, probably?

Reply:
No, the callers rely on this to signify a race.